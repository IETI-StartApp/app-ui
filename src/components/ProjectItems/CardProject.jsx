import React from 'react';
import {Avatar, Box, Card, CardContent, CardHeader, CardMedia, Typography} from '@material-ui/core';
import {useStyles} from './styles';
import ProgressGraph from './ProgressGraph';
import MonetizationOnIcon from '@material-ui/icons/MonetizationOn';

export default function CardProject({project}) {
    const { name, image, finance} = project;
    const {value, valuation, startDate} = finance;
    const classes = useStyles();
    const owner = "Nombre de usuario";

    return (
        <Card className={classes.sizeCard}>
            <CardHeader
                avatar={
                    <Avatar aria-label="photo" src={image}>
                    </Avatar>
                }
                title={owner}
                subheader={startDate}
            />
            <CardMedia
                component="img"
                alt="Project image"
                height="210"
                image={image}
            />
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    {name}
                </Typography>
                <Typography variant="body2" component="p">
                    $ {valuation} Meta de financiaci√≥n
                </Typography>
                <Box position="relative" display="inline-flex" mt={2}>
                    <Box mr={5}>
                        <Box position="relative" display="inline-flex">
                            <Avatar className={classes.avatar}><MonetizationOnIcon/></Avatar>
                        </Box>
                        <Typography> Financiado </Typography>
                        <Typography className={classes.mainColor}> $ {value}</Typography>
                    </Box>
                    <Box mr={5}>
                        <ProgressGraph progress={value * 100 / valuation}/>
                        <Typography> Progreso </Typography>
                    </Box>
                </Box>
            </CardContent>
        </Card>
    );
}
