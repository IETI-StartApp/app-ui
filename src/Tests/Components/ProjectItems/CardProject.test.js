import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import React from 'react';
import CardProject from "../../../../src/components/ProjectItems/CardProject";
import {CardHeader, CardMedia, Typography} from '@material-ui/core';

describe('Testing card project', () => {

    const wrapper = shallow(<CardProject project={{
        "country": "CO",
        "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lacus laoreet non curabitur gravida. Vel turpis nunc eget lorem. Commodo sed egestas egestas fringilla. Est placerat in egestas erat.",
        "finance": {
            "value": 10000,
            "investorNumber": 1,
            "valuation": 50000,
            "minimumInvestment": 1,
            "startDate": {"$date": "2019-02-16T05:00:00.000Z"},
            "endDate": {"$date": "2019-02-16T05:00:00.000Z"}
        },
        "image": "https://images.unsplash.com/photo-1619537765439-1469ac26014a?ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1400&q=80",
        "name": "Beach & talk",
        "video": "https://youtu.be/hCrJTLdmkHo"
    }}/>)

    test('should load CardProject', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should load a valid Card Header', () => {
        const cardHeader = wrapper.find(CardHeader).at(0);

        expect(cardHeader.props().title).toBeDefined();
        expect(cardHeader.props().subheader).toBeDefined();
        expect(cardHeader.props().avatar).toBeDefined();

        expect(cardHeader.props().title).not.toEqual("");
        expect(cardHeader.props().subheader).not.toEqual("");
    })

    test('should load the project image', () => {
        const image = wrapper.find(CardMedia).at(0).props().image;
        expect(image).toBeDefined();
        expect(image).not.toEqual("");
    })

    test('should load a valids Typographys', () => {
        const projectName = wrapper.find(Typography).at(0).text();
        expect(projectName).toBeDefined();
        expect(projectName).not.toEqual("");

        const fObjective = wrapper.find(Typography).at(2).text().split(" Meta")[0];
        expect(fObjective).not.toBeNaN();

        const financed = wrapper.find(Typography).at(4).text();
        expect(financed.substring(4, financed.length - 1)).not.toBeNaN();
        expect(parseFloat(financed.substring(4, financed.length - 1))).toBeNaN()
    })

})
