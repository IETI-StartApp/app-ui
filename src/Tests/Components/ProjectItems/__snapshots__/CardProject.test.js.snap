// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing card project should load CardProject 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-sizeCard-1"
  variant="outlined"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <WithStyles(ForwardRef(Avatar))
        aria-label="photo"
        src="https://images.unsplash.com/photo-1619537765439-1469ac26014a?ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1400&q=80"
      />
    }
    subheader={
      Object {
        "$date": "2019-02-16T05:00:00.000Z",
      }
    }
    title="Nombre de usuario"
  />
  <WithStyles(ForwardRef(CardMedia))
    alt="Project image"
    component="img"
    height="210"
    image="https://images.unsplash.com/photo-1619537765439-1469ac26014a?ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1400&q=80"
  />
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        Beach & talk
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        component="p"
        variant="body2"
      >
        Meta de financiaci√≥n: $ 
        50000
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Grid))
        alignItems="start"
        className="makeStyles-cardGrid-4"
        container={true}
        direction="row"
        justify="center"
        spacing={5}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <WithStyles(ForwardRef(Avatar))
              className="makeStyles-avatar-3"
            >
              <Memo([object Object]) />
            </WithStyles(ForwardRef(Avatar))>
          </div>
          <WithStyles(ForwardRef(Typography))
            align="center"
          >
             Financiado 
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            align="center"
            className="makeStyles-mainColor-2"
          >
             $ 
            10000
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <ProgressGraph
              progress={20}
            />
          </div>
          <WithStyles(ForwardRef(Typography))
            align="center"
          >
             Progreso 
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <WithStyles(ForwardRef(Avatar))
              className="makeStyles-avatar-3"
            >
               
              5
               
            </WithStyles(ForwardRef(Avatar))>
          </div>
          <WithStyles(ForwardRef(Typography))
            align="center"
          >
             Socios 
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
</WithStyles(ForwardRef(Card))>
`;
